AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -357,8 +430,35 @@ - (void)_deleteNodesAtIndexPaths:(NSArray *)indexPaths withAnimationOptions:(ASD
   [self deleteNodesOfKind:ASDataControllerRowNodeKind atIndexPaths:indexPaths completion:^(NSArray *nodes, NSArray *indexPaths) {
     ASDisplayNodeAssertMainThread();
     
+    // TODO - Filter out Move indexPaths
+    NSMutableArray *newNodes = [[NSMutableArray alloc] initWithCapacity:nodes.count];
+    NSMutableArray *newIndexPaths = [[NSMutableArray alloc] initWithCapacity:indexPaths.count];
+    
+    for (int i = 0; i < indexPaths.count; i++) {
+      if (![_moveFromIndexPaths containsObject:indexPaths[i]]) {
```

Comment:
`_moveFromIndexPaths` is modified on the editing queue, but it's being read here from the main queue. Would it make sense to put these data structures inside the `_ASHierarchyChangeSet` and then pass that object into these methods as a `context` object? 

That would cut down on the state inside the data controller.